daml-lf/validation/src/main/scala/com/digitalasset/daml/lf/validation/Typing.scala

```diff
@@ -358,7 +358,9 @@ private[validation] object Typing {
               throw EIllegalShadowingExprVar(ctx, param)
             checkExpr(controllers, TParties)
           }
-          checkExpr(observers, TParties) // NICK, should this be conditional on supportsContractObservers?
+          observers.foreach {
+            checkExpr(_, TParties) // NICK, should this be conditional on supportsContractObservers?
+          }
```

Comment:
```suggestion
          observers.foreach(checkExpr(_, TParties))
```
Don't need the comment anymore.